File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {


    @Reference
    private BrandService brandService;

    /**
     * 查询所有
     * @return
     */
    @RequestMapping("/findAll")
    @ResponseBody
    public List<TbBrand> finidAll(){

        return brandService.findAll();
    };

    /**
     * 分页查询
     * @param page
     * @param rows
     * @return
     */

    @RequestMapping("/findPage")
    public PageResult findPage (int page ,int rows){

        return brandService.findPage(page, rows);
    }

    /**
     * 添加数据
     * @param tbBrand
     * @return
     */
    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand tbBrand){

        try {

            brandService.save(tbBrand);
            return  new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    /**
     * 删除数据
     * @param ids
     * @return
     */

    @RequestMapping("/delete")
    public Result delete(Long[] ids){

        try {
            brandService.delete(ids);
           return   new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    /**
     * 根据id查询数据
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(long id){


         return  brandService.findOne(id);

    }

    /**
     * 修改数据
     * @param tbBrand
     * @return
     */

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){

        try {

            brandService.update(tbBrand);
            return  new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 条件查询
     * @param tbBrand
     * @return
     */
    @RequestMapping("/select")
    public PageResult select(@RequestBody TbBrand tbBrand,int page ,int rows){

      return  brandService.findPage(tbBrand,page,rows);

    }
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
    return  brandService.selectOptionList();

    }

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope, brandService,$controller) {

    $controller("baseController",{$scope:$scope});//继承


    //查询全部数据
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {

                $scope.list = response;
            }
        );
    }

    //定义分页方法
    $scope.findPage = function (page, rows) {
        brandService.findPage(page, rows).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据

                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }
    //保存方法
    $scope.save = function () {
        //定定义变量，如果id值为空，表示是新增数据，用save方法，如果不为空，表示为修改数据，调用update方法
        var Object = brandService.save($scope.entity);
        if ($scope.entity.id != null) {
            Object = brandService.update($scope.entity);
        }
        Object.success(
            function (response) {

                if (response.success) {

                    $scope.reloadList();

                } else {
                    alert(response.message);
                }
            }
        );
    }

    // 删除方法

    $scope.dele = function () {

        if(confirm('确定要删除吗？')){
        brandService.dele($scope.selectids).success(
            function (responde) {
                if (responde.success) {

                    $scope.reloadList();

                } else (
                    alert(responde.message)
                );
            }
        );
    };
    }
    //根据id查询
    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        );
    }

    $scope.searchEntity = {};
    // 条件查询
    $scope.search = function (page, rows) {
        brandService.search(page, rows, $scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );

    }
    $scope.findBrandList=function () {
        brandService.selectOptionList().success(
            function (response) {
                $scope.entity=response;
            }
        )
    }

});
